MoltenVK/MoltenVK/GPUObjects/MVKQueue.mm

```diff
@@ -366,15 +358,22 @@
 		}
 #if MVK_XCODE_12
 		if (mvkGetMVKConfiguration()->debugMode) {
-			MVKLogInfo("Shader log messages:");
+			bool isFirstMsg = true;
 			for (id<MTLFunctionLog> log in mtlCB.logs) {
+				if (isFirstMsg) {
+					MVKLogInfo("Shader log messages:");
```

Comment:
So that's why this was crashing when I didn't use the device to log. I remember changing it to use the device, but I made that change (and the initial change) on another computer, and I excitedly made a patch to copy back in order to submit it *before* I made those changes, so the change I submitted here was still broken.

Reply:
Yeah. `MVKLogInfo()` calls the `reportMessage()` function on the current object, so it needs to be alive.